File: src/main/java/net/matees/itemrush/settings/MaxItemCount.java
package net.matees.itemrush.settings;

import net.matees.Setting;

public class MaxItemCount implements Setting<Integer> {

    private Integer maxItemCount = 64;

    @Override
    public String getName() {
        return "Max Item Count";
    }

    @Override
    public Integer getSetting() {
        return maxItemCount;
    }

    public void setSetting(Integer setting) {
        this.maxItemCount = setting;
    }
}

File: src/main/java/net/matees/arcade/itemrush/settings/MaxItemCount.kt
package net.matees.arcade.itemrush.settings

import net.matees.settings.AbstractIntegerSetting
import org.bukkit.Material

class MaxItemCount private constructor() : AbstractIntegerSetting() {
    private var maxItemCount = 1

    override val name: String
        get() = "Max Item Count"

    override val description: String
        get() = "Max amount of items dropped (if random is enabled, this is the range)"

    override var setting: Int?
        get() = maxItemCount
        set(setting) {
            if (setting != null) {
                this.maxItemCount = setting
            }
        }

    override val menuItemMaterial: Material
        get() = Material.STICK

    override val menuItemSlot: Int
        get() = 12

    companion object {
        val instance: MaxItemCount = MaxItemCount()
    }
}
